### ISSUE TITLE
False negative: multiple spaces around reserved words not detected

### ISSUE THREAD

**Issue by Tieshiyang** [checkstyle/checkstyle#17725]
I have read check documentation: 
https://checkstyle.sourceforge.io/checks/whitespace/whitespaceafter.html
https://checkstyle.sourceforge.io/checks/whitespace/whitespacearound.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

From: https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace
> Beyond where required by the language or other style rules, and apart from literals, comments and Javadoc, a single ASCII space also appears in the following places only.
• Separating any reserved word, such as if, for or catch, from an open parenthesis (() that follows it on that line.
• Separating any reserved word, such as else or catch, from a closing curly brace (}) that precedes it on that line.

Code: 
```java
class Test {
  void testIf(int x) {
    if   (x > 0) { // Expected 1 Violation for multiple spaces
      System.out.println("Positive");
    }       else       { // Expected 2 Violations for multiple spaces
      return;
    }
  }

  void testFor() {
    for    (int i = 0; i < 5; i++) { // Expected 1 Violation for multiple spaces
      System.out.println(i);
    }
  }
}
```

Cli: 
```cmd
$ java -jar checkstyle-10.26.1-all.jar -c google_checks.xml Test.java 
Starting audit...
Audit done.
```

Formatter:
```cmd
$ java -jar google-java-format-1.28.0-all-deps.jar Test.java > FormattedCode.java
$ diff -Naru Test.java FormattedCode.java
--- Test.java   2025-09-02 00:32:50.285539400 +0800
+++ FormattedCode.java  2025-09-02 00:33:04.442512300 +0800
@@ -1,16 +1,15 @@
 class Test {
   void testIf(int x) {
-    if   (x > 0) { // Expected 1 Violation for multiple spaces
+    if (x > 0) { // Expected 1 Violation for multiple spaces
       System.out.println("Positive");
-    }       else       { // Expected 2 Violations for multiple spaces
+    } else { // Expected 2 Violations for multiple spaces
       return;
     }
   }

   void testFor() {
-    for    (int i = 0; i < 5; i++) { // Expected 1 Violation for multiple spaces
+    for (int i = 0; i < 5; i++) { // Expected 1 Violation for multiple spaces
       System.out.println(i);
     }
   }
-
 }
```

---

According to the style guide, there must be exactly one space between reserved words (`if`, `for`) and the following `(`, and also between `}` and `else`.
However, Checkstyle reports nothing when there are multiple spaces in these positions.
This is a false negative: the formatter correctly normalizes them, but the linter is silent.

**Suggestion**: Add a property (e.g., `spaceNum=1`) to `WhitespaceAfter` / `WhitespaceAround` (ideally `WhitespaceBefore`) to enforce exactly one space.
This would make Checkstyle consistent with the formatter and the style guide.

---

**Comment by romani** [comment]
We should not add new property to Check, but we should have something to catch multiple spaces for sure, we can make Google style specific Check to catch all that required.

**Comment by romani** [comment]
This issue might be duplicate of https://github.com/checkstyle/checkstyle/issues/17756 but let's keep both open for now

**Comment by romani** [comment]
@Krish-876 , should we add usage of https://checkstyle.sourceforge.io/checks/whitespace/singlespaceseparator.html to cover what user wants ?

**Comment by Krish-876** [comment]
> [@Krish-876](https://github.com/Krish-876) , should we add usage of https://checkstyle.sourceforge.io/checks/whitespace/singlespaceseparator.html to cover what user wants ?

I don’t know if `SingleSpaceSeparatorCheck` is the best fit here, but from my understanding it enforces spacing globally rather than on specific tokens.

From what I can see, the request is about enforcing exactly one space after specific reserved words (per the Google Style Guide). Since `SingleSpaceSeparatorCheck` is a FileSetCheck and works on the whole file, I guess it could also flag spacing in contexts like alignment or comments, which the user didn’t intend.

I think this kind of rule fits better in a TreeWalker check such as `WhitespaceAroundCheck`, which already targets these reserved-word tokens and would allow the spac

**Comment by romani** [comment]
SingleSpaceSeparator  is TreWalker child - https://checkstyle.sourceforge.io/checks/whitespace/singlespaceseparator.html#Parent_Module , so it is not global.

**Comment by Krish-876** [comment]
Thanks for the clarification,
I was mainly thinking about the difference between text based behavior and token specific enforcement. Happy to go with whatever approach fits best here.

**Comment by romani** [comment]
Let's try existing Check and in testing we will see if it has blocker side effects

**Comment by Youssefalaa7** [comment]
I am on it

**Comment by romani** [comment]
fix is done by our hero @Youssefalaa7 .
